File: Microservicos Angular/angular/ucis/src/app/services/history-assignment.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Bed } from '../models/bed';
import { HistoryAssignment } from '../models/history-assignment';
import { Patient } from '../models/patient';
import { CommonService } from './common.service';
import { END_POINT_BASE } from '../config/app';

@Injectable({
  providedIn: 'root',
})
export class HistoryAssignmentService extends CommonService<HistoryAssignment> {
  protected urlEndPoint: string = END_POINT_BASE + '/beds/historyAssignments';

  constructor(http: HttpClient) {
    super(http);
  }

  findAllPatients(): Observable<Patient[]> {
    return this.http.get<Patient[]>(`${this.urlEndPoint}/patients`);
  }
  findAllBeds(): Observable<Bed[]> {
    return this.http.get<Bed[]>(`${this.urlEndPoint}/beds`);
  }
}

File: microservice-beds/src/main/java/edu/ucentral/microservice/beds/service/HistoryAssignmentService.java
package edu.ucentral.microservice.beds.service;

import java.util.List;

import edu.ucentral.commons.beds.model.Bed;
import edu.ucentral.commons.beds.model.HistoryAssignment;
import edu.ucentral.commons.patients.model.Patient;
import edu.ucentral.commons.service.services.CommonService;

public interface HistoryAssignmentService extends CommonService<HistoryAssignment> {

	List<Patient> findAllPatients();

	List<Bed> findAllBeds();
	// HistoryAssignment findBedByPatientId(Long id);

}
